torchmetrics/utilities/checks.py

```diff
@@ -557,13 +550,33 @@ def _check_retrieval_inputs(
     if indexes.dtype is not torch.long:
         raise ValueError("`indexes` must be a tensor of long integers")
 
+    _check_retrieval_target_and_prediction_types(preds, target, allow_non_binary_target=allow_non_binary_target)
+
+    return indexes.long().flatten(), preds.float().flatten(), target.long().flatten()
+
+
+def _check_retrieval_target_and_prediction_types(
```

Comment:
instead of creating a new function, why not let this be part of the original `_check_retrieval_inputs` function?